pkg/cmd/ban.go

```diff
@@ -70,25 +68,23 @@ func CreateBannedUser(ctx *clicontext.CommandContext, userSignupName string, con
 		return err
 	}
 
-	bannedUser, err := newBannedUser(userSignup, ksctlConfig.Name)
+	bannedUser, err := banneduser.NewBannedUser(userSignup, ksctlConfig.Name)
 	if err != nil {
 		return err
 	}
 
-	emailHashLabelMatch := runtimeclient.MatchingLabels(map[string]string{
-		toolchainv1alpha1.BannedUserEmailHashLabelKey: bannedUser.Labels[toolchainv1alpha1.BannedUserEmailHashLabelKey],
-	})
-	bannedUsers := &toolchainv1alpha1.BannedUserList{}
-	if err := cl.List(context.TODO(), bannedUsers, emailHashLabelMatch, runtimeclient.InNamespace(cfg.OperatorNamespace)); err != nil {
+	if err := ctx.PrintObject(userSignup, "UserSignup to be banned"); err != nil {
 		return err
 	}
 
-	if err := ctx.PrintObject(userSignup, "UserSignup to be banned"); err != nil {
+	alreadyBannedUser, err := banneduser.GetBannedUser(ctx, bannedUser.Labels[toolchainv1alpha1.BannedUserEmailHashLabelKey], cl, cfg.OperatorNamespace)
+	if err != nil {
```

Comment:
It's a small detail but before these changes we printed the UserSignup (`ctx.PrintObject(userSignup, ...`) after listing banned users because if there was an error listing the banned users then we returned with the error message before printing the entire UserSignup. Did you change the order intentionally?

Reply:
Unintentionally, thanks! Fixed